client/src/partials/inventory/list/modals/actions.modal.js

```diff
@@ -0,0 +1,106 @@
+angular.module('bhima.controllers')
+.controller('InventoryListActionsModalController', InventoryListActionsModalController);
+
+InventoryListActionsModalController.$inject = [
+  'AccountService', 'InventoryService', 'InventoryGroupService',
+  'InventoryTypeService', 'InventoryUnitService', 'NotifyService',
+  '$uibModalInstance', 'Store', 'data'
+];
+
+function InventoryListActionsModalController(Account, Inventory, InventoryGroup, InventoryType, InventoryUnit, Notify, Instance, Store, Data) {
+  var vm = this, session = vm.session = {};
+
+  // variables for storage
+  var GroupStore = {},
+      TypeStore  = {},
+      UnitStore  = {};
+
+  // map for actions
+  var map = { 'add' : addList, 'edit' : editList };
+
+  // expose to the view
+  vm.submit = submit;
+  vm.cancel = cancel;
+
+  // startup
+  startup();
+
+  /** submit data */
+  function submit(form) {
+    if (form.$invalid) {
+      return;
+    }
+
+    var record = Inventory.clean(vm.session);
+    map[vm.action](record, vm.identifier)
+    .then(function (res) {
+      Instance.close(res);
+    });
+  }
+
+  /** add inventory list */
+  function addList(record) {
+    return Inventory.create(record)
+    .then(function (res) {
```

Comment:
These two are equivalent:
```js
function addList(record) {
  return Inventory.create(record)
    .then(function (res) {
       return res;
     })
     .catch(Notify.errorHandler);
}
```
... compared to ...
```js
function addList(record) {
  return Inventory.create(record)
    .catch(Notify.errorHandler);
}
```

The second version is clearer, uses less code (easier to read, understand, and maintain), and is less likely to be poorly written and cause a mistake.  Could you use it for your promises instead of `return res;`?